src/page-analysis/background/make-screenshot.js

```diff
@@ -1,17 +1,52 @@
 import delay from 'src/util/delay'
 import { whenPageLoadComplete, whenTabActive } from 'src/util/tab-events'
+import { dataURLToBlob } from 'blob-util'
 
 // Take a screenshot of the tabId, if it is active.
 // Returns a promise of the screenshot (a png image in a data URL).
 // The promise rejects if the tab is not currently active!
 async function snapNow({ tabId }) {
     const tab = await browser.tabs.get(tabId)
-    const image = await browser.tabs.captureVisibleTab(tab.windowId, {
-        format: 'png',
+    let image = await browser.tabs.captureVisibleTab(tab.windowId, {
+        format: 'jpeg',
     })
+    const blob = await dataURLToBlob(image)
+    console.log('size before ' + blob.size / 1000 + 'kb')
+    image = await resizeImage(image, 0.75, 200)
+
     return image
 }
 
+async function resizeImage(image, scale, maxHeight) {
+    return new Promise((resolve, reject) => {
+        let img = new Image()
+        let canvas = document.createElement('canvas')
+        let ctx = canvas.getContext('2d')
+
+        img.src = image
+        img.onload = () => {
+            let newHeight = Math.floor(img.height * scale)
+
+            if (newHeight <= maxHeight) {
```

Comment:
If this condition is not met (i.e., 75% of the screenshot height > 200px), when will the Promise ever resolve? Wouldn't most screenshots on full windows be way taller than that?

Haven't really done much with `<canvas>` before, so probably just me missing something.

Reply:
@poltak On `line 42`. `img.src = canvas.toDataUrl()` resets the img and loops through the `img.onload` process again reducing it until it is the size we want.

Reply:
That makes sense. So it will keep scaling down and re-triggering the event until it's below `maxHeight`. Cool stuff!